### PR TITLE
Improve Registry Experience & Simplify Context Values

### PR DESCRIPTION
**Pull request by alexyaang**
- [x] Make `registryExperience()` better so that users only get prompted if a the command is possible on a certain registry provider
- [x] Simplify context values https://github.com/microsoft/vscode-docker/pull/4026#discussion_r1313331522

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -492,7 +492,7 @@
                 },
                 {
                     "command": "vscode-docker.registries.copyImageDigest",
-                    "when": "view == dockerRegistries && viewItem =~ /registryV2Tag/i",
+                    "when": "view == dockerRegistries && viewItem =~ /commontag/i && !(viewItem =~ /commontag;dockerhub/i)",
```
Comment (bwateratmsft) [inline package.json:495]:
```suggestion
                    "when": "view == dockerRegistries && viewItem =~ /commontag/i && !(viewItem =~ /commontag;.*dockerhub/i)",
```

Comment (bwateratmsft) [reply package.json:495]: This applies also to line 515, 520, 525, 530, 540

Comment (alexyaang) [reply package.json:495]: looks good. we currently know that there isn't any context value between commontag and dockerhub, so is this for if we add something in the future?

Comment (bwateratmsft) [reply package.json]: Yep, exactly

**Review by bwateratmsft — APPROVED** [review]

**Pull request merged**